### PR TITLE
endpoints for user's keys

### LINKED ISSUE #8227: [Server] - Endpoint to return keys mapped to a user
#### Current Behavior

We have an endpoint in meshery cloud that returns keys associated with a user. `api/identity/users/keys`. This api is not supported in meshery ui. 

keys
api

#### Desired Behavior

We want the same api to be supported inside meshery ui so that the permissions framework [PR](https://github.com/meshery/meshery/pull/7859) can be merged 

#### Mockups

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Freedisch**
**Notes for Reviewers**

This PR fixes #8227
Add endpoint for to retrieve keys from user
`api/identity/users/keys`

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Chadha93** [comment]
@Freedisch  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

File: server/models/providers.go [hunk 1]
```diff
@@ -256,6 +256,8 @@ const (
 	UsersProfile Feature = "users-profile"
 
 	UsersIdentity Feature = "users-identity"
+
+	UsersKeys Feature = "" //To be filled...
```
Comment (theBeginner86) [inline server/models/providers.go:260]:
```suggestion
	UsersKeys Feature = "users-keys"
```

Comment (theBeginner86) [reply server/models/providers.go:260]: I have added the necessary support to expose this capability in remote provider. 👍

**Comment by leecalcote** [comment]
@Freedisch has this API endpoint been added to the "Meshery API Endpoints" spreadsheet?

File: server/handlers/user_handler.go [hunk 2]
```diff
@@ -80,6 +80,25 @@ func (h *Handler) GetUsers(w http.ResponseWriter, req *http.Request, _ *models.P
 	fmt.Fprint(w, string(resp))
 }
 
+func (h *Handler) GetUsersKeys(w http.ResponseWriter, req *http.Request, _ *models.Preference, _ *models.User, provider models.Provider) {
```
Comment (leecalcote) [inline server/handlers/user_handler.go:83]: API documentation needed here.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: server/handlers/user_handler.go (hunk 2 above)
Comment (Freedisch) [reply server/handlers/user_handler.go:100]: done ✅

**Comment by Freedisch** [comment]
@leecalcote, In the spreadsheet I saw a couple of endpoints that were not yet implemented like
`/api/meshmodel/<any-endpoint>?ver={version}` to `Filter resource by version` 
`/api/meshmodels/models/{model}/components` to `Returns a list of registered components within model scope. ` and more...

Would it be okay if I go ahead and implement them or it's not necessary

**Comment by abdullah1308** [comment]
@Freedisch can you paste a screenshot of the response ?

File: server/handlers/user_handler.go [hunk 3]
```diff
@@ -80,6 +80,42 @@ func (h *Handler) GetUsers(w http.ResponseWriter, req *http.Request, _ *models.P
...
+//
+// ```?page={page-number}``` Default page number is 0
+//
+// ```?pagesize={pagesize}``` Default pagesize is 20
+// 
+// ```?search={id|function|owner|category}``` If search is non empty then a greedy search is performed
+//
+// ```?filter={condition}```
+// responses:
+// 	200: keys
```
Comment (abdullah1308) [inline server/handlers/user_handler.go:98]: This response needs to defined under `server/handlers/doc.go`.

File: server/handlers/user_handler.go [hunk 4]
```diff
@@ -80,6 +80,42 @@ func (h *Handler) GetUsers(w http.ResponseWriter, req *http.Request, _ *models.P
 	fmt.Fprint(w, string(resp))
 }
 
+// swagger:route GET /api/identity/users/keys UserKeysAPI idGetAllUsersKeysHandler
```
Comment (abdullah1308) [inline server/handlers/user_handler.go:83]: Run 
```
make swagger-build
```

To update the swagger files with these new docs

**Review by abdullah1308 — CHANGES_REQUESTED** [review]

File: server/handlers/user_handler.go (hunk 4 above)
Comment (leecalcote) [reply server/handlers/user_handler.go:83]: Output:

```
 make swagger-docs-build
swagger generate spec -o ./docs/_data/swagger.yml --scan-models; \
	swagger flatten ./docs/_data/swagger.yml -o ./docs/_data/swagger.yml --with-expand --format=yaml
object has no key "keys"
make: *** [swagger-docs-build] Error 1
```

**Comment by Freedisch** [comment]
@abdullah1308, 
image

File: server/handlers/user_handler.go (hunk 3 above)
Comment (abdullah1308) [reply server/handlers/user_handler.go:98]: @leecalcote @Freedisch You are unable to run `make swagger-docs-build` as keys isn't defined under `server/handlers/doc.go`

**Comment by abdullah1308** [comment]
@Freedisch Is the Meshery Cloud endpoint for keys paginated ? If it is, you need to add pagination here as well.

**Comment by Freedisch** [comment]
> @Freedisch Is the Meshery Cloud endpoint for keys paginated ? If it is, you need to add pagination here as well.

No, it's not paginated

**Comment by Freedisch** [comment]
@abdullah1308, updated the docs

**Comment by theBeginner86** [comment]
I'm tying this off now...

**Review by theBeginner86 — APPROVED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit 1d677d0535e402cb93ae59c3d956201f02b41c45 at: https://64d8f4512fce3c2453194531--meshery-docs-preview.netlify.app

**Pull request merged**